src/session-pool.ts

```diff
@@ -425,10 +425,9 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
    * @param {GetReadSessionCallback} callback The callback function.
    */
   getReadSession(callback: GetReadSessionCallback): void {
-    this._acquire(types.ReadOnly).then(
-      session => callback(null, session),
```

Comment:
Errors generated by `callback(null, session)` are not caught by the second parameter provided to `then()`. If they are intended to be caught by `callback`,  this must be done in a separate `catch()` block

Reply:
I'm not very familiar with how this code is intended to work, but it seems we would want to execute `callback` once. If executing `callback` in the success case throws an error, calling it again with its own error seems like a round trip that we could skip. From your investigation, would there be another way to return the error when it happens, or is this the best way to handle it?

Thanks for all of your help!

Reply:
@stephenplusplus I am not familiar either with how it's supposed to work; so it might not be the best way to solve this issue. 

I decided to re-use the `callback` parameter because it was a function with error handling capabilities and it was available where the error occurred. 
Re-using `callback` possibly adds an extra round trip; but most Node.js callbacks handle potential errors first; before doing anything else; so I assume it wouldn't lead to too much code being run for nothing.

However, it might make more sense to handle errors generated by `callback` inside `callback` itself; but I wasn't sure where it was defined; and what was the best way to handle errors.

Maybe we should add a `try/catch` block [around this line ?](https://github.com/googleapis/nodejs-spanner/blob/23958ae48f49306cf38755831db091fef16998fb/src/database.ts#L2109)